src/cpuinfo_ppc.c

```diff
@@ -0,0 +1,359 @@
+DECLARE_SETTER(PPCFeatures, ebb)
+DECLARE_SETTER(PPCFeatures, isel)
+DECLARE_SETTER(PPCFeatures, tar)
+DECLARE_SETTER(PPCFeatures, vcrypto)
+DECLARE_SETTER(PPCFeatures, htm_nosc)
+DECLARE_SETTER(PPCFeatures, arch300)
+DECLARE_SETTER(PPCFeatures, ieee128)
+DECLARE_SETTER(PPCFeatures, darn)
+DECLARE_SETTER(PPCFeatures, scv)
+DECLARE_SETTER(PPCFeatures, htm_no_suspend)
+
+static const CapabilityConfig kConfigs[] = {
+    {{PPC_FEATURE_32, 0}, "ppc32", &set_ppc32},
+    {{PPC_FEATURE_64, 0}, "ppc64", &set_ppc64},
+    {{PPC_FEATURE_601_INSTR, 0}, "ppc601", &set_ppc601},
+    {{PPC_FEATURE_HAS_ALTIVEC, 0}, "altivec", &set_altivec},
+    {{PPC_FEATURE_HAS_FPU, 0}, "fpu", &set_fpu},
+    {{PPC_FEATURE_HAS_MMU, 0}, "mmu", &set_mmu},
+    {{PPC_FEATURE_HAS_4xxMAC, 0}, "4xxmac", &set_mac_4xx},
+    {{PPC_FEATURE_UNIFIED_CACHE, 0}, "ucache", &set_unifiedcache},
+    {{PPC_FEATURE_HAS_SPE, 0}, "spe", &set_spe},
+    {{PPC_FEATURE_HAS_EFP_SINGLE, 0}, "efpsingle", &set_efpsingle},
+    {{PPC_FEATURE_HAS_EFP_DOUBLE, 0}, "efpdouble", &set_efpdouble},
+    {{PPC_FEATURE_NO_TB, 0}, "notb", &set_no_tb},
+    {{PPC_FEATURE_POWER4, 0}, "power4", &set_power4},
+    {{PPC_FEATURE_POWER5, 0}, "power5", &set_power5},
+    {{PPC_FEATURE_POWER5_PLUS, 0}, "power5+", &set_power5plus},
+    {{PPC_FEATURE_CELL, 0}, "cellbe", &set_cell},
+    {{PPC_FEATURE_BOOKE, 0}, "booke", &set_booke},
+    {{PPC_FEATURE_SMT, 0}, "smt", &set_smt},
+    {{PPC_FEATURE_ICACHE_SNOOP, 0}, "ic_snoop", &set_icachesnoop},
+    {{PPC_FEATURE_ARCH_2_05, 0}, "arch_2_05", &set_arch205},
+    {{PPC_FEATURE_PA6T, 0}, "pa6t", &set_pa6t},
+    {{PPC_FEATURE_HAS_DFP, 0}, "dfp", &set_dfp},
+    {{PPC_FEATURE_POWER6_EXT, 0}, "power6x", &set_power6ext},
+    {{PPC_FEATURE_ARCH_2_06, 0}, "arch_2_06", &set_arch206},
+    {{PPC_FEATURE_HAS_VSX, 0}, "vsx", &set_vsx},
+    {{PPC_FEATURE_PSERIES_PERFMON_COMPAT, 0}, "archpmu", &set_pseries_perfmon_compat},
+    {{PPC_FEATURE_TRUE_LE, 0}, "true_le", &set_truele},
+    {{PPC_FEATURE_PPC_LE, 0}, "ppcle", &set_ppcle},
+    {{0, PPC_FEATURE2_ARCH_2_07}, "arch_2_07", &set_arch207},
+    {{0, PPC_FEATURE2_HTM}, "htm", &set_htm},
+    {{0, PPC_FEATURE2_DSCR}, "dscr", &set_dscr},
+    {{0, PPC_FEATURE2_EBB}, "ebb", &set_ebb},
+    {{0, PPC_FEATURE2_ISEL}, "isel", &set_isel},
+    {{0, PPC_FEATURE2_TAR}, "tar", &set_tar},
+    {{0, PPC_FEATURE2_VEC_CRYPTO}, "vcrypto", &set_vcrypto},
+    {{0, PPC_FEATURE2_HTM_NOSC}, "htm-nosc", &set_htm_nosc},
+    {{0, PPC_FEATURE2_ARCH_3_00}, "arch_3_00", &set_arch300},
+    {{0, PPC_FEATURE2_HAS_IEEE128}, "ieee128", &set_ieee128},
+    {{0, PPC_FEATURE2_DARN}, "darn", &set_darn},
+    {{0, PPC_FEATURE2_SCV}, "scv", &set_scv},
+    {{0, PPC_FEATURE2_HTM_NO_SUSPEND}, "htm-no-suspend", &set_htm_no_suspend},
+};
+static const size_t kConfigsSize = sizeof(kConfigs) / sizeof(CapabilityConfig);
+
+static bool HandlePPCLine(const LineResult result,
+                              PPCPlatformStrings* const strings) {
```

Comment:
Do you mind running `git clang-format`?

Reply:
How do you have clang-format set up? When I try to run it I get 
```
Error reading /home/rashmica/working/cpu_features/.clang-format: Invalid argument
YAML:4:4: error: unknown key '...'
...
   ^

```
I've pushed the changes that this config gave me - figured it should be close to what you're looking for?
``` clang-format -style=google -dump-config > .clang-format```

Reply:
It may be because of a missing line feed at the end of `.clang-format`. I submitted a fix, can you patch the PR and try again?

Reply:
Yup, no issues running git clang-format now.